tests/model_fields_/test_embedded_model.py

```diff
@@ -104,6 +113,46 @@ def test_nested(self):
         )
         self.assertCountEqual(Book.objects.filter(author__address__city="NYC"), [obj])
 
+    def truncate_ms(self, value):
+        """Truncate microsends to millisecond precision as supported by MongoDB."""
+        return value.replace(microsecond=(value.microsecond // 1000) * 1000)
+
+    def test_ordering_by_embedded_field(self):
+        query = Holder.objects.filter(data__integer__gt=3).order_by("-data__integer").values("pk")
+        expected = [{"pk": e.pk} for e in list(reversed(self.objs[4:]))]
+        self.assertSequenceEqual(query, expected)
+
+    def test_ordering_grouping_by_embedded_field(self):
+        expected = sorted(
+            (Holder.objects.create(data=Data(integer=x)) for x in range(6)),
+            key=lambda x: x.data.integer,
+        )
+        query = (
+            Holder.objects.annotate(
+                group=ExpressionWrapper(F("data__integer") + 5, output_field=IntegerField())
+            )
+            .values("group")
+            .annotate(max_auto_now=Max("data__auto_now"))
+            .order_by("data__integer")
+        )
+        query_response = [{**e, "max_auto_now": self.truncate_ms(e["max_auto_now"])} for e in query]
+        self.assertSequenceEqual(
+            query_response,
+            [
+                {"group": e.data.integer + 5, "max_auto_now": self.truncate_ms(e.data.auto_now)}
+                for e in expected
+            ],
+        )
+
+    def test_ordering_grouping_by_sum(self):
```

Comment:
Again, it's hard for me to reason about the expected result of query, so some explanatory comment would be useful. For example, the "cheat sheet" at https://docs.djangoproject.com/en/dev/topics/db/aggregation/ is easy to follow.